.github/workflows/validate_source_protos.yml

```diff
@@ -0,0 +1,25 @@
+name: Validate Source Proto Files
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  validate:
+    name: Validate Source Proto Files
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v2
+
+    - name: Setup python3
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.7.x'
+
+    - name: Install build dependencies
+      run: python -m pip install -r python_build_requirements.txt
+
+    - name: Validate source proto files against generated
+      run: |
+        python source/codegen/validate_source_protos.py -s source/codegen/protos -g generated
```

Comment:
Since the script only grabs the proto files, you could technically put these .proto files in the same directories as the other metadata files, and make it easier on us when copying from hapigen exports.

Reply:
Yeah, I thought about that and thought it was weird having the .proto files alongside the metadata files. So my thought was to have them in this separate folder here and also update the .mak files in hapigen to output the .proto files to a "proto" folder in the exports as well.

I could also just update this submission to put the .proto files next to the metadata files in the source/codegen/metadata folder instead.

Thoughts?

Reply:
Talked with Alex and he was okay with putting the .proto files alongside the metadata files. For now, that'll be source/codegen/metadata/<driver_name>. I went ahead and moved the proto files there and updated the script/workflow to look there.

Long term, we want any imported metadata/proto files to be somewhere other than source. Perhaps in the imports directory. That applies to both hapigen and scrapigen sourced metadata/proto files. But for now, keeping it in souce/codegen/metadata will be alright.